Excel-custom-functions/Batching/README.md

```diff
@@ -0,0 +1,80 @@
+1. On the machine where your custom functions project is installed, follow the instructions to install the self-signed certificates (https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md) . 
+2. From a command prompt from within your custom functions project directory, run `npm run start` to start a localhost server instance. 
+3. Run `npm run sideload` to launch Excel and load the custom functions add-in. Additonal information on sideloading can be found at <https://aka.ms/sideload-addins>.
+4. After Excel launches, you will need to register the custom-functions add-in to work around a bug:
+    a. On the upper-left-hand side of Excel, there is a small hexagon icon with a dropdown arrow. The icon is to right of the Save icon.
+    b. Click on this dropdown arrow and then click on the Custom Functions Sample add-in to register it.
+5. Test a custom function by entering `=CONTOSO.ADD(num1, num2)` in a cell.
+6. Try the other functions in the sample: `=CONTOSO.ADDASYNC(num1, num2)`, `CONTOSO.INCREMENTVALUE(increment)`.
+7. If you make changes to the sample add-in, copy the updated files to your website, and then close and reopen Excel. If your functions are not available in Excel, re-insert the add-in using **Insert** > **My Add-ins**.
+
+## Making changes
+If you make changes to the sample functions code (in the JS file), close and reopen Excel to test them.
+
+If you change the functions metadata (in the JSON file), close Excel and delete your cache folder `Users/<user>/AppData/Local/Microsoft/Office/16.0/Wef/CustomFunctions`. Then re-insert the add-in using **Insert** > **My Add-ins**.
+
+## Debugging
+Currently, the best method for debugging Excel custom functions is to first [sideload](https://docs.microsoft.com/office/dev/add-ins/testing/sideload-office-add-ins-for-testing) your add-in within **Excel Online**. Then you can debug your custom functions by using the [F12 debugging tool native to your browser](https://docs.microsoft.com/office/dev/add-ins/testing/debug-add-ins-in-office-online). Use `console.log` statements within your custom functions code to send output to the console in real time.
+
+If your add-in fails to register, [verify that SSL certificates are correctly configured](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md) for the web server that's hosting your add-in application.
+
+If you are testing your add-in in Office 2016 desktop you can enable [runtime logging](https://docs.microsoft.com/office/dev/add-ins/testing/troubleshoot-manifest#use-runtime-logging-to-debug-your-add-in) to debug issues with your add-in's XML manifest file as well as several installation and runtime conditions.
+
+## IntelliSense for the JSON file in Visual Studio Code	
+For intelliSense to help you edit the JSON file, follow these steps:
+
+1. Open the JSON file (it has a .json extension) in Visual Studio Code.	
+2. If you are starting a new file from scratch, add the following to the top of the file:	
```

Comment:
Use contractions: You are should be you're (check elsewhere too, I think I noticed a couple others.)